### DISCUSSION TITLE
Arduino Mega 2560

### DISCUSSION THREAD

**Discussion by andre2007** [denizzzka/d_c_arm_test#17] [Q&A]
Do you know by chance whether the approach you used here would also be possible for Arduino Mega 2560? Or is BetterC the only usable approach here?

**Comment by denizzzka** [comment]
upd:
Current druntime is about 200Kb, so maybe, but it is need to implement backend

**Reply by andre2007** [reply]
Thanks for clarification.

**Reply by denizzzka** [reply]
There is one more circumstance: our language is declared as language for 32- and 64-bit systems, not for 16 bit.

**Reply by andre2007** [reply]
As BetterC seems to work (https://wiki.dlang.org/D_on_AVR) my assumption was this is more a performance issue than a technical issue.

I mean when I use `int` in cases when `short` or `byte` is sufficient.

**Comment by andre2007** [comment]
> it is need to implement backend

Is my understanding correct that you mean that in your DRuntime fork some `Arduino Mega 2560` specific coding is missing and needs to be implemented?

**Reply by denizzzka** [reply]
Yes. I implemented only ARM Cortex-M3 backend (and it also can be used for Cortex-M4) with FreeRTOS interconnection to provide multithreading.

This backend stored here: https://github.com/denizzzka/d_c_arm_test/tree/master/d/freertos_druntime_backend/external
